daemons/gptp/common/common_port.cpp

```diff
@@ -467,8 +467,18 @@ bool CommonPort::processStateChange( Event e )
 
 bool CommonPort::processSyncAnnounceTimeout( Event e )
 {
+	// We're Grandmaster, set grandmaster info to me
+	ClockIdentity clock_identity;
+	unsigned char priority1;
+	unsigned char priority2;
+	ClockQuality clock_quality;
+
+	Timestamp system_time;
+	Timestamp device_time;
+	uint32_t local_clock, nominal_clock_rate;
+
 	// Nothing to do
-	if( clock->getPriority1() != 255 )
+	if( clock->getPriority1() == 255 )
```

Comment:
Yes, this is called when the timer for timeouts based on *reception* of SYNC/ANNOUNCE frames goes off. This path is here to allow us to consider taking over being grand master if the current one goes away, but if we are not capable of doing so, there's no point in trying to. It might still be a good idea to have a status log saying "Hey, our GM apparently went away but I can't take over because priority1 is 255."